docs/sources/http_api/data_source.md

```diff
@@ -712,3 +714,93 @@ Content-Type: application/json
   }
 }
 ```
+
+## Query a data source by ID
+
+Queries a data source having backend implementation.
+
+`POST /api/ds/query`
+
+> **Note:** Most of Grafana's builtin data sources have backend implementation.
+
+**Example request for the Test data source**:
+
+```http
+POST /api/ds/query HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+
+{
+   "queries":[
+      {
+         "refId":"A",
+         "scenarioId":"csv_metric_values",
+         "datasource":{
+            "uid":"PD8C576611E62080A",
+            "type":"testdata"
+         },
+         "stringInput":"1,20,90,30,5,0",
+         "datasourceId":6,
```

Comment:
Is this really here? And/or should we render it here now when we have the datasource ref above?

Reply:
Yep that's here alright. It's a legacy field AFAICT that gets auto set. Will try dig a bit further on it and see if it can be removed without breaking other things

Reply:
As discussed, since datasource ref is included above we can leave this out in the example since we want users to migrate away from using `datasourceId`.